src/main/java/uk/gov/justice/digital/delius/transformers/NsiTransformer.java

```diff
@@ -29,9 +54,25 @@ public NsiTransformer (@Autowired final RequirementTransformer requirementTransf
                 .actualStartDate(n.getActualStartDate())
                 .expectedStartDate(n.getExpectedStartDate())
                 .referralDate(n.getReferralDate())
+                .length(n.getLength())
+                .lengthUnit(NSI_LENGTH_UNIT)
+                .nsiManagers(nsiManagersOf(n.getNsiManagers()))
+                .court(courtTransformer.courtOf(nsi.getEvent().getCourt()))
```

Comment:
This feels odd, be interestingly to know why the frontend needs the event court (which I'm not sure what that is, the sentencing the court maybe??)

Maybe worth checking this and explaining the thinking around here it "just doesn't feel right"

Reply:
Ok, I will try and get some clarification